data/data/gcp/dns/base.tf

```diff
@@ -0,0 +1,51 @@
+resource "google_dns_managed_zone" "int" {
+  name       = "${var.cluster_id}-int"
+  dns_name   = "${var.cluster_domain}."
```

Comment:
do we have guarantees from the API that it will not have `.`

Reply:
The . at the end is required on create. Passing a dns name without it results in a 400 invalid dns name error.

Reply:
Bu t what if cluster_domain already has a trailing `.` ?

Reply:
ahh, ic what you mean, will fix

Reply:
@abhinavdahiya so in the terraform vars file we say we don't expect a trailing dot:
https://github.com/openshift/installer/blob/master/data/data/config.tf#L25-L32
However, when we validate install-config we do allow a trailing dot:
https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L59
Added a commit to ensure that what we pass to terraform doesn't have one.

Reply:
https://github.com/openshift/installer/pull/1977/commits/f9c69fc4c2620ebb67a8a35ec8883b52f50ceff6

doesn't solve the problem that `cluster_domain` doesn't have trailing `.` suffix

Reply:
ok *now* it's fixed